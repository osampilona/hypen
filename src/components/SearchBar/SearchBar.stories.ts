import type { Meta, StoryFn } from "@storybook/react";
import SearchBar, { ISearchBarProps } from "@/components/SearchBar/SearchBar";

const meta: Meta<typeof SearchBar> = {
  title: "Atomic Components/Search Bar",
  component: SearchBar,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
};

export default meta;

const Template: StoryFn<ISearchBarProps> = (args) => SearchBar({ ...args });

export const Default = Template.bind({});
Default.args = {
  labelWhat: "What",
  labelWhere: "Where",
  labelWhen: "When",
};
