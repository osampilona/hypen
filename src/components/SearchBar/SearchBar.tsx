import React from "react";
import searchBar from "@/components/SearchBar/searchBar.module.scss";
import FilterIcon from "@/icons/FilterIcon";
import SearchIcon from "@/icons/SearchIcon";
import LocationIcon from "@/icons/LocationIcon";
import CalendarIcon from "@/icons/CalendarIcon";

export interface ISearchBarProps {
  labelWhat: string;
  labelWhere: string;
  labelWhen: string;
}

const SearchBar = (props: ISearchBarProps) => {
  return (
    <div className={searchBar.container}>
      <div className={searchBar.search}>
        <div className={searchBar.icon}>
          <SearchIcon />
          <p>{props.labelWhat}</p>
        </div>
        <div className={searchBar.separator}></div>
        <div className={searchBar.icon}>
          <LocationIcon />
          <p>{props.labelWhere}</p>
        </div>
        <div className={searchBar.separator}></div>
        <div className={searchBar.icon}>
          <CalendarIcon />
          <p>{props.labelWhen}</p>
        </div>
      </div>
      <div className={searchBar.icon}>
        <FilterIcon />
      </div>
    </div>
  );
};

export default SearchBar;
