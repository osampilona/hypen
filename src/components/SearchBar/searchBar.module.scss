.container {
  border-radius: 1.5625rem;
  border: 0.3px solid rgba(69, 69, 69, 0.3);
  background: #fff;
  box-shadow: 1px 0px 15px 2px rgba(40, 40, 40, 0.05);
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0.78rem 1.5rem;
  color: #686868;
  gap: 2rem;
  font-family: system-ui;

  .search {
    display: flex;
    align-items: center;
    gap: 12px;
  }

  .icon {
    display: flex;
    align-items: center;
    gap: 10px;
    svg {
      fill: #706868;
      height: 20px;
      width: auto;
    }
  }

  .separator {
    width: 0.05rem;
    height: 1.125rem;
    background-color: #e9e9e9;
    opacity: 0.8;
  }
}
