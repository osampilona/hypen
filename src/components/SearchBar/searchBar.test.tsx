import { describe, test, expect } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import SearchBar from "@/components/SearchBar/SearchBar";

describe("SearchBar", () => {
  test("initially, overlay is not visible", () => {
    render(<SearchBar />);
    expect(screen.queryByTestId("overlay-container")).not.toBeTruthy();
  });

  test("clicking on 'What' button toggles overlay visibility and updates state", () => {
    render(<SearchBar />);

    // Simulate click on "What" button
    fireEvent.click(screen.getAllByTestId("searchBarButtonWhat")[1]);
    expect(screen.getByTestId("overlay-container")).toBeTruthy();
    expect(screen.getByPlaceholderText("What")).toBeTruthy();

    // Simulate click on "What" button again to toggle off
    fireEvent.click(screen.getAllByTestId("searchBarButtonWhat")[1]);
    expect(screen.queryByTestId("overlay-container")).not.toBeTruthy();
  });

  test("clicking on 'Where' button toggles overlay visibility and updates state", () => {
    render(<SearchBar />);

    // Simulate click on "Where" button
    fireEvent.click(screen.getAllByTestId("searchBarButtonWhere")[2]);
    expect(screen.getByTestId("overlay-container")).toBeTruthy();
    expect(screen.getByPlaceholderText("Where")).toBeTruthy();

    // Simulate click on "Where" button again to toggle off
    fireEvent.click(screen.getAllByTestId("searchBarButtonWhere")[2]);
    expect(screen.queryByTestId("overlay-container")).not.toBeTruthy();
  });

  test("clicking on 'When' button toggles overlay visibility and updates state", () => {
    render(<SearchBar />);

    // Simulate click on "When" button
    fireEvent.click(screen.getAllByTestId("searchBarButtonWhen")[3]);
    expect(screen.getByTestId("overlay-container")).toBeTruthy();
    expect(screen.getByTestId("custom-calendar")).toBeTruthy();

    // Simulate click on "When" button again to toggle off
    fireEvent.click(screen.getAllByTestId("searchBarButtonWhen")[3]);
    expect(screen.queryByTestId("overlay-container")).not.toBeTruthy();
  });

  test("closing overlay sets state correctly", () => {
    render(<SearchBar />);

    // Simulate click on "What" button to show overlay
    fireEvent.click(screen.getAllByTestId("searchBarButtonWhat")[1]);
    expect(screen.getByTestId("overlay-container")).toBeTruthy();
    expect(screen.getByPlaceholderText("What")).toBeTruthy();

    // Simulate closing the overlay
    fireEvent.click(screen.getByTestId("overlay-container"));
    expect(screen.queryByTestId("overlay-container")).not.toBeTruthy();
  });

  test("handles input change for 'What' field", () => {
    render(<SearchBar />);

    // Simulate click on "What" button to show overlay
    fireEvent.click(screen.getAllByTestId("searchBarButtonWhat")[1]);
    const input = screen.getByPlaceholderText("What");

    // Simulate input change
    fireEvent.change(input as HTMLInputElement, { target: { value: "test" } });
    expect((input as HTMLInputElement).value).toBe("test");
  });

  test("handles input change for 'Where' field", () => {
    render(<SearchBar />);

    // Simulate click on "Where" button to show overlay
    fireEvent.click(screen.getAllByTestId("searchBarButtonWhere")[2]);
    const input = screen.getByPlaceholderText("Where");

    // Simulate input change
    fireEvent.change(input as HTMLInputElement, {
      target: { value: "test location" },
    });
    expect((input as HTMLInputElement).value).toBe("test location");
  });
});
