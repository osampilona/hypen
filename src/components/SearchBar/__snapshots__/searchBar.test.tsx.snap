// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBar > handleClick updates isLabelClicked state correctly 1`] = `
<div>
  <div
    class="_container_528985"
  >
    <div
      class="_search_528985"
    >
      <div
        class="_icon_528985"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.25 2a8.25 8.25 0 0 1 6.34 13.53l5.69 5.69a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-5.69-5.69A8.25 8.25 0 1 1 10.25 2ZM3.5 10.25a6.75 6.75 0 1 0 13.5 0 6.75 6.75 0 0 0-13.5 0Z"
          />
        </svg>
        <p
          class="_label_528985 "
          data-testid=""
        >
          What
        </p>
      </div>
      <div
        class="_separator_528985"
      />
      <div
        class="_icon_528985"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 48c-79.5 0-144 61.39-144 137 0 87 96 224.87 131.25 272.49a15.77 15.77 0 0 0 25.5 0C304 409.89 400 272.07 400 185c0-75.61-64.5-137-144-137z"
            fill="none"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="32"
          />
          <circle
            cx="256"
            cy="192"
            fill="none"
            r="48"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="32"
          />
        </svg>
        <p
          class="_label_528985 "
          data-testid=""
        >
          Where
        </p>
      </div>
      <div
        class="_separator_528985"
      />
      <div
        class="_icon_528985"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.75 0a.75.75 0 0 1 .75.75V3h9V.75a.75.75 0 0 1 1.5 0V3h2.75c.966 0 1.75.784 1.75 1.75v16a1.75 1.75 0 0 1-1.75 1.75H3.25a1.75 1.75 0 0 1-1.75-1.75v-16C1.5 3.784 2.284 3 3.25 3H6V.75A.75.75 0 0 1 6.75 0ZM21 9.5H3v11.25c0 .138.112.25.25.25h17.5a.25.25 0 0 0 .25-.25Zm-17.75-5a.25.25 0 0 0-.25.25V8h18V4.75a.25.25 0 0 0-.25-.25Z"
          />
        </svg>
        <p
          class="_label_528985 "
          data-testid=""
        >
          When
        </p>
      </div>
    </div>
    <div
      class="_icon_528985"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.75 12.75h1.5a.75.75 0 0 0 0-1.5h-1.5a.75.75 0 0 0 0 1.5ZM12 6a.75.75 0 0 1 .75-.75h7.5a.75.75 0 0 1 0 1.5h-7.5A.75.75 0 0 1 12 6ZM12 18a.75.75 0 0 1 .75-.75h7.5a.75.75 0 0 1 0 1.5h-7.5A.75.75 0 0 1 12 18ZM3.75 6.75h1.5a.75.75 0 1 0 0-1.5h-1.5a.75.75 0 0 0 0 1.5ZM5.25 18.75h-1.5a.75.75 0 0 1 0-1.5h1.5a.75.75 0 0 1 0 1.5ZM3 12a.75.75 0 0 1 .75-.75h7.5a.75.75 0 0 1 0 1.5h-7.5A.75.75 0 0 1 3 12ZM9 3.75a2.25 2.25 0 1 0 0 4.5 2.25 2.25 0 0 0 0-4.5ZM12.75 12a2.25 2.25 0 1 1 4.5 0 2.25 2.25 0 0 1-4.5 0ZM9 15.75a2.25 2.25 0 1 0 0 4.5 2.25 2.25 0 0 0 0-4.5Z"
        />
      </svg>
    </div>
  </div>
</div>
`;
